Clarify more record fetching

mattias-p opened the pull request:
Clarify record fetching for Dnssec03-05, Syntax06 and Zone06. This PR attempts to clarify the specification in a way that's consistent with the implementation.

matsduf commented on docs/specifications/tests/DNSSEC-TP/dnssec04.md:
```diff
@@ -23,15 +23,18 @@ The domain name to be tested.
+2. Create a DNSKEY query for the zone with DO flag set.
+3. Create a SOA query for the zone with DO flag set.
+4. Send the DNSKEY query over UDP to one of the name server IP addresses.
+5. Send the SOA query over UDP to one of the name server IP addresses.
```
```
4b. If the name server does not reply or does not reply with the AA bit set, pick the next
IP from the list in 1 and rerun 4 with that IP. If there are no more IPs to use, this test case
fails.

5. Send the SOA query over UDP to the same name server IP address as above.

5b. Extract the RRSIG records from the responds. If any of the responds lack RRSIG 
records this test case ends with fail.
```

matsduf commented on docs/specifications/tests/DNSSEC-TP/dnssec05.md:
```diff
@@ -19,10 +19,12 @@ The domain name to be tested.
-2. The DNSKEY algorithm numbers are derived from all the DNSKEY RRs.
-3. The algorithm numbers are matched with the IANA DNSKEY algorithm table.
-4. If any the algorithm numbers is unassigned, reserved, a private algorithm
+1. Obtain a set of name server IP addresses using [Method4] and [Method5].
```
```
1. Obtain a list of name server IP addresses using [Method4] and [Method5].
```

matsduf commented on docs/specifications/tests/DNSSEC-TP/dnssec05.md:
```diff
@@ -19,10 +19,12 @@ The domain name to be tested.
-3. The algorithm numbers are matched with the IANA DNSKEY algorithm table.
-4. If any the algorithm numbers is unassigned, reserved, a private algorithm
+1. Obtain a set of name server IP addresses using [Method4] and [Method5].
+2. Create a DNSKEY query for the zone with DO flag set.
```
```
2. Create a DNSKEY query with DO flag set for the apex of the child zone .
```

matsduf commented on docs/specifications/tests/DNSSEC-TP/dnssec05.md:
```diff
@@ -19,10 +19,12 @@ The domain name to be tested.
-4. If any the algorithm numbers is unassigned, reserved, a private algorithm
+1. Obtain a set of name server IP addresses using [Method4] and [Method5].
+2. Create a DNSKEY query for the zone with DO flag set.
+3. Send the DNSKEY query over UDP to one the name server IP addresses.
```
3b. If the name server does not reply or does not reply with the AA bit set, pick the next
IP from the list in 1 and rerun 3 with that IP. If there are no more IPs to use, this test case
fails.

mattias-p commented:
@matsduf I'd like to keep this PR focused on what queries are sent where. I've taken care not to touch parts that look at the responses. I haven't even wrapped the lines in those parts. The only reason those lines are touched by this PR is because I had to bump the numbering in the ordered list. Please make your own PR for changes you want to make in those parts.

mattias-p commented on docs/specifications/tests/DNSSEC-TP/dnssec03.md:
```diff
@@ -39,16 +39,21 @@ The domain name to be tested.
-6. Depending on the key sizes listed in section 10.3 of RFC 5155, if the
-   number of iterations is larger than the corresponding key size of
-   the smallest key, this test case fails.
+ 1. Obtain a set of name server IP addresses using [Method4] and [Method5].
```
Are you suggesting changing the term "set" to the term "list"? Since we're uninterested in both duplicates and order, the term "set" is a better fit.

matsduf replied:
Yes, but on the other hand, "next" fits list better.

mattias-p replied:
I don't understand what "next" you're referring to.

matsduf replied:
In my proposal I had "if no answer go to next IP". Now I cannot see that because you have updated.

mattias-p replied:
I updated the query sending step to account for retrying. I did not add any details on what's to be done when the set is exhausted.

mattias-p commented on docs/specifications/tests/DNSSEC-TP/dnssec03.md:
```diff
@@ -39,16 +39,21 @@ The domain name to be tested.
+ 6. The iterations value is extracted from the NSEC3PARAM RR.
+ 7. If the number of iterations is higher than 100, this test case emits
+    at least a warning.
+ 8. Send the DNSKEY query over UDP to one of the name server IP addresses.
```
This PR is an attempt to clarify the specification in a way that's consistent with the implementation. Your suggestion is contradicted by the current implementation so it is in fact a change request. Please create a new issue about changing the specification and behavior in synchrony.

matsduf commented on docs/specifications/tests/Syntax-TP/syntax06.md:
```diff
@@ -17,17 +17,20 @@ The domain name to be tested.
+3. Send the SOA query over UDP to each unique name server IP address.
+4. Get the RNAME from the SOA record.
+5. Convert the first non-escaped "." to an "@" in the RNAME.
+6. De-escape the RNAME, converting a "\." to ".".
```
Another way of writing 5 and 6:

* Create an empty string.
* Add the first label to the string, if any.
* Exit converting to mail address if RNAME consists of zero or one labels.
* Append an "@" to the string.
* For remaining labels in RNAME:
   * Append the label.
   * Append a "." unless the label was appended.

matsduf commented on docs/specifications/tests/Zone-TP/zone06.md:
```diff
@@ -30,12 +30,14 @@ The domain name to be tested.
-3. Retrieve the SOA minimum value from the SOA record.
-4. If the minimum value is larger than 86400 seconds (1 day), this test
+1. Obtain a set of name server IP addresses using [Method4] and [Method5].
+2. Create a SOA query for the zone.
```
```
2. Create a SOA query for the apex of the zone.
```

mattias-p commented:
I added the agreed upon language in the remaining places.
Additionally I added a commit with the same language for the Syntax TCs as well, since I realized that they're also only checking the first response they get.
